docs/guides/js-waku/message-encryption.md

```diff
@@ -0,0 +1,218 @@
+
+// Retrieve messages from Store peers
+await node.store.queryWithOrderedCallback([decoder], callback);
+```
+
+## Signing encrypted messages
+
+Message signing helps in proving the authenticity of received messages. By attaching a signature to a message, you can verify its origin and integrity with absolute certainty.
+
+The `sigPrivKey` option allows the `Symmetric` and `ECIES` message `encoders` to sign the message before encryption using an `ECDSA` private key:
+
+```js
+import { generatePrivateKey } from "@waku/message-encryption";
+import { createEncoder as createSymmetricEncoder } from "@waku/message-encryption/symmetric";
+import { createEncoder as createECIESEncoder } from "@waku/message-encryption/ecies";
+
+// Generate a random private key for signing messages
+const sigPrivKey = generatePrivateKey();
+
+// Create a symmetric encoder that signs messages
+const symmetricEncoder = createSymmetricEncoder({
+	contentTopic: contentTopic, // message content topic
+	symKey: symKey, // symmetric key for encrypting messages
+	sigPrivKey: sigPrivKey, // private key for signing messages before encryption
+});
+
+// Create an ECIES encoder that signs messages
+const ECIESEncoder = createECIESEncoder({
+	contentTopic: contentTopic, // message content topic
+	publicKey: publicKey, // ECIES public key for encrypting messages
+	sigPrivKey: sigPrivKey, // private key for signing messages before encryption
+});
+
+// Send and receive your messages as usual with Light Push and Filter
+await subscription.subscribe([symmetricEncoder], callback);
+await node.lightPush.send(symmetricEncoder, { payload });
+
+await subscription.subscribe([ECIESEncoder], callback);
+await node.lightPush.send(ECIESEncoder, { payload });
+```
+
+You can extract the `signature` and its public key (`signaturePublicKey`) from the [DecodedMessage](https://js.waku.org/classes/_waku_message_encryption.DecodedMessage.html) and compare it with the expected public key to verify the message:
+
+```js
+// Generate a random private key for signing messages
+const sigPrivKey = generatePrivateKey();
+
+// Generate a public key from the private key for verifying signatures
+const sigPubKey = getPublicKey(sigPrivKey);
+
+// Create an encoder that signs messages
+const encoder = createEncoder({
+	contentTopic: contentTopic,
+	symKey: symKey,
+	sigPrivKey: sigPrivKey,
+});
+
+// Modify the callback function to verify message signature
+const callback = (wakuMessage) => {
+	// Extract the message signature and public key of the signature
+	const signature = wakuMessage.signature;
+	const signaturePublicKey = wakuMessage.signaturePublicKey;
+
+	// Compare the public key of the message signature with the sender's own
+	if (JSON.stringify(signaturePublicKey) === JSON.stringify(sigPubKey)) {
+		console.log("This message was correctly signed");
+	} else {
+		console.log("This message has an incorrect signature");
+	}
+};
+```
+
+## Restoring encryption keys
```

Comment:
I am not sure this is necessary//useful.

However, do feel free to use https://github.com/waku-org/js-waku-examples/tree/master/examples/eth-pm/src/key_pair_handling as an example on how to save/restore key information from local storage using proper security (Subtle Crypto)

Reply:
I've added the `key_pair_handling` example, but I still think the `bytesToHex` and `hexToBytes` examples are helpful for some key exporting use cases. Let me know what you think about the update